src/client/api/tests/job_tests.c

```diff
@@ -73,7 +73,7 @@ int uname(struct utsname *buf)
 		return -1;
 	}
 
-	strncpy(buf->nodename, uname_nodename, _UTSNAME_LENGTH);
+	strncpy(buf->nodename, uname_nodename, _UTSNAME_LENGTH - 1);
```

Comment:
My 'C' is a bit rusty, but what does `uname_nodename` point at a this point?  Doesn't it need to be 65 bytes of `malloc()`ed memory?  I don't see anywhere in this file where that is done.

Reply:
uname_nodename is the source string, and it's being statically allocated in the individual tests. It's always null terminated, and in practice unlikely to be longer than the 65 bytes.

Reply:
I.e. if it was longer, it would be a developer error. The tests are totally controlled by us.

Reply:
Oh right.  I always forget the argument order is dest, src.  So totally unnatural.